content/guides/concepts/dht.md

```diff
@@ -0,0 +1,85 @@
+
+### Peer lists
+
+Peers also keep connection to other peers to forward requests if the requested hash is not in their own bucket.
+
+If hashes are of length n, a peer will keep n-1 lists of peers: 
+- the first list contains peers which ID have a different 1st bit.
+- the second list contains peer which have their first bits identical to its own, but a different second bit
+- ...
+- the m-th list contains peer which have their first m-1 bits identical, but a different m-th bit
+- ...
+
+The higher m, the harder it is to find peers which have the same ID up to m bits. The lists of "closest" peers typically remains empty.
+"Close" here is defined as the XOR distance, so the longer the prefix they share, the closer they are.
+Lists also have a maximum of entries k, otherwise the first lists would contain half the network, then a fourth, etc.
+
+### Using the DHT
+
+When a peer receives a lookup request, it will either answer with a value if it falls into its own bucket, or answer with the contacting information (IP + port, peerID, etc) of a closer peer. The requesting peer can then send its request to this closer peer. The process goes on until a peer is able to answer it. 
+A request for a hash of length n will take at maximum log2(n) steps, or even log2m(n). 
+
+# The DHT of IPFS
+
+In IPFS Kademlia's DHT, keys are SHA256 hashes.
+[PeerIDs](https://docs.libp2p.io/concepts/peer-id/) are those of [libp2p](https://libp2p.io/), the networking library used by IPFS.
+
+We use a DHT to lookup two types of objects (both represented by a SHA256):
+- [Content IDs](https://docs.ipfs.io/guides/concepts/cid/) of the data added to IPFS. A lookup of this value will give the peerIDs of the peers having this immutable content
+- [IPNS records](https://docs.ipfs.io/guides/concepts/ipns/). A lookup will give the last Content ID associated with this IPNS address, enabling routing mutable content
+
+Consequently, IPFS's DHT is one of the way of mutable and immutable [Content Routing](https://docs.libp2p.io/concepts/content-routing/). It's currently the only one [implemented](https://libp2p.io/implementations/#peer-routing). 
+
+Per specification, the default bucket size k is 20 meaning each of the 255 lists of peers contain at most 20 peers. 
+
+## Usage
+
+### Adding an entry
+
+Adding a blob of data to IPFS is just advertizing that you have it. Since DHT is the only content routing implemented, you can just use
+` ipfs add myData`
+IPFS will automatically chunk your data and add a mapping on the DHT between the Content ID and your PeerID. Note that there can be other Peer IDs already mapped to that value, so you will be added to the list. Also note that if the provided data is bigger than 124kb, it will be chunked in "blocks" and both blocks and overall data will be mapped. 
```

Comment:
```suggestion
IPFS will automatically chunk your data and add a mapping on the DHT between the Content ID and your `peerID`. Note that there can be other `peerID`s already mapped to that value, so you will be added to the list. Also note that if the provided data is bigger than 124KB, it will be chunked into "blocks", and both those blocks and the overall data will be mapped. 
```

Reply:
Moved this change to https://github.com/ipfs/docs/pull/364